### ISSUE TITLE
Cost Explorer API (5/n): refinement of future Cost Explorer API related work

### ISSUE THREAD

**Issue by consideRatio** [2i2c-org/infrastructure#4677]
Represents refinment of future work for the Cost Explorer (CE) API, as part of #4453.

- [x] Discuss with Yuvi about 2i2c:hub-name tag not capturing much costs (~30 USD / month out of ~1k) in a Erik/Yuvi 1on1
  Since openscapes isn't tagged well, we are to piece together information based on whats available. For future accounts with better tagging, another approach can possibly be used.
- [x] terraform and eksctl: try to ensure tags for new projects are applied to all our resources created via terraform and eksctl
- [ ] aws billing: overview most costs from 2i2c infra, figure out how to isolate them in openscapes specifically, and then also how we would identify them in new projects
- [ ] code: Cache requests to CE to reduce costs (0.01 per request) and improve performance
- [ ] chart: Declare resource requests and limits
- [ ] chart: Add and validate NetworkPolicy resources deployed so that _only Grafana_ can work against the webserver
- [ ] chart: Use gunicorn instead of flask to run the flask application

## Requirements for this task to be considered refined

- The list of followup work is populated based on knowledge gained after completing the issues named Cost Explorer API (1-4/n)
- The list is carefully considered to be include things that directly moves us towards a MVP of sorts, and doesn't just contain optimizations.

## Definition of done

- A reasonable amount of items from the list of potential followup work defined and refined as issues into the epic issue, this should include mostly MVP driving issues and not focus on optimizations etc.
- Any items not defined and refined are added to the list as non-issue items into the epic issue

**Comment by yuvipanda** [comment]
Thanks @consideRatio. As guidance for scope of tasks, please use the EPIC and create separate subtasks for each items than using a 'catch all'. The goal is to make sure we have things that can be sized during next sprint planning, and single catch-all tasks can not be. Sized tasks help us figure out team velocity to know how much we can deliver. This also means there are *refined things to mark as comitted next time you're working*, which was one of the issues this sprint. 

So the specific ask is:

1. Use #4453 to track, rather than a catchall issue like this.
2. Create issues for each task (with just enough refinement for tracking progress), so we can look at what needs to be done by looking at the EPIC
3. Spend some time explicitly and intentionally refining remining tasks on the EPIC, now that you have more information. This is to 'bootstrap' refined tasks for you to pick up on next sprint, so you don't have to do unplanned work.

**Comment by consideRatio** [comment]
Sorry for the confusion @yuvipanda, I'm aligned with that intent.

I perceived it was a bit premature to define/refine individual tasks from this list now, that I mostly wanted as a overview of various things to define/refine.

In the spirit of how we are to do things, this issue can be seen as the task of doing the work of defining+refining misc followup tasks to the first four initially defined, ending with #4673. This refinment issue should itself not be considered refined until the list of followup tasks is defined.

I'll update it to reflect that it should be the task of defining+refining followup work after #4673.

**Comment by yuvipanda** [comment]
> I perceived it was a bit premature to define/refine individual tasks from this list now, that I mostly wanted as a overview of various things to define/refine.

@consideRatio that makes sense. I'd like you to use #4453 for that, rather than creating another issue. Can you move the tasklist here to that?

> This refinment issue should itself not be considered refined until the list of followup tasks is defined.

+1, I think using this issue to say 'refine further' is appropriate.

**Comment by consideRatio** [comment]
This refinement effort is now complete!
